src/Access/ContactFormsAccessCheck.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_contact_forms\Access;
+
+use Drupal\contact\ContactFormInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Access\AccessResultInterface;
+
+/**
+ * Route access handler for contact form routes.
+ */
+class ContactFormsAccessCheck implements AccessInterface {
+
+  /**
+   * Checks access for corporate contact forms based on allow_canonical_url.
+   *
+   * @param Drupal\contact\ContactFormInterface $contact_form
+   *   The contact form to check against.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(ContactFormInterface $contact_form = NULL): AccessResultInterface {
+    // Deny access if form is corporate and the 'Allow canonical URL' is FALSE.
+    if (
```

Comment:
You are missing cache metadata here.

Reply:
lets discuss

Reply:
cache dependency on the contact form and on the route context